exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/ApiControllerTest.java

```diff
@@ -104,23 +104,21 @@ void tearDown(VertxTestContext context) {
 
   @Test
   void submitValidTransaction(VertxTestContext context) {
-    String messageJson = "{\"service_id\":42}";
+    BinaryMessage message = getBinaryMessage();
+
     String messageHash = "1234";
-    BinaryMessage message = mock(BinaryMessage.class);
     Transaction transaction = mock(Transaction.class);
 
-    when(jsonBinaryMessageConverter.toMessage(messageJson))
-        .thenReturn(message);
-    when(service.convertToTransaction(message))
+    when(service.convertToTransaction(any(BinaryMessage.class)))
```

Comment:
Can we use `eq(message)` here?

Reply:
I think no, our test message and ApiController deserialized messages are not the same.

Reply:
I see, MessageReader does not implement equals, and as the whole hierarchy of Messages is about to be replaced with another one (hopefully, a simpler one), it doesn't make sense to implement e/h at the moment.